AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/SyncMutationToCloudOperation.swift

```diff
@@ -179,6 +181,17 @@ class SyncMutationToCloudOperation: Operation {
         }
     }
 
+    private func resolveReachabilityPublisher(request: GraphQLRequest<MutationSync<AnyModel>>) {
+        if networkReachabilityPublisher == nil {
+            guard let apiName = request.apiName else {
+                return
```

Comment:
from what i remember, GraphQLRequest can be created without `apiName` and so depending on the context in which API Plugin is called (either on the GraphQL APIS or the Rest APIs), it will traverse through the endpointConfig at runtime to try and infer the single endpoint that the caller is trying to call, so to provide an experience where caller doesn't need to specify the apiName. 

My thought for this guard statement and then returning when it cannot find the apiName: we know datastore calls api plugin currently without apiName and happy path is that there's only one endpoint configured. is there a missing piece here for getting the reachability for apiName = nil?